src/uproot/behaviors/TBranch.py

```diff
@@ -609,31 +609,60 @@ def dask(
 
     filter_branch = uproot._util.regularize_filter(filter_branch)
 
-    if open_files:
-        return _get_dask_array(
-            files,
-            filter_name,
-            filter_typename,
-            filter_branch,
-            recursive,
-            full_paths,
-            step_size,
-            custom_classes,
-            allow_missing,
-            real_options,
-        )
+    if library.name == "np":
```

Comment:
We want the code in TBranch.py to avoid special cases that depend on the value of `library`; that's what `library.*` methods are for. Can this be refactored to call `library.something` here, and then have the different `library` implementations do the different things?